src/validator.rs

```diff
@@ -1548,6 +1549,56 @@ impl<'a> Validator<'a> {
         }
     }
 
+    /// Expect the block to contain any number of `key = value` fields
+    /// where each key is a unique Item of type itype.
+    /// Run the closure `f(key, vd)` for every matching block.
+    #[cfg(feature = "vic3")]
+    pub fn validate_item_key_values<F>(&mut self, itype: Item, mut f: F)
+    where
+        F: FnMut(&Token, ValueValidator),
+    {
+        let mut visited_fields = HashMap::new();
```

Comment:
This should use TigerHashMap (from crate::helpers) for consistency with other hashmaps.